injector/src/main/java/edu/ucr/cs/riple/injector/changes/AddSingleElementAnnotation.java

```diff
@@ -0,0 +1,150 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package edu.ucr.cs.riple.injector.changes;
+
+import com.github.javaparser.ast.NodeList;
+import com.github.javaparser.ast.expr.AnnotationExpr;
+import com.github.javaparser.ast.expr.ArrayInitializerExpr;
+import com.github.javaparser.ast.expr.Expression;
+import com.github.javaparser.ast.expr.Name;
+import com.github.javaparser.ast.expr.SingleMemberAnnotationExpr;
+import com.github.javaparser.ast.expr.StringLiteralExpr;
+import com.github.javaparser.ast.nodeTypes.NodeWithAnnotations;
+import com.google.common.base.Preconditions;
+import edu.ucr.cs.riple.injector.location.Location;
+import java.util.Optional;
+
+/**
+ * Used to add <a
+ * href="https://docs.oracle.com/javase/specs/jls/se8/html/jls-9.html#jls-SingleElementAnnotation">Single
+ * Element Annotations</a> with only one member on elements in source code. If the annotation
+ * already exists, it can be collapsed into a single annotation with multiple elements if requested.
+ */
+public class AddSingleElementAnnotation extends AddAnnotation {
+  /** Argument of the annotation. */
+  private final String argument;
+
+  /**
+   * If true, if an annotation with the same name exists, the existing annotation should include the
+   * passed argument, otherwise, a new annotation will be injected to the node.
+   */
+  private final boolean collapseArguments;
+
+  public AddSingleElementAnnotation(
+      Location location, String annotation, String argument, boolean collapseArguments) {
+    super(location, annotation);
+    Preconditions.checkArgument(
+        argument != null && !argument.equals(""),
+        "argument cannot be null or empty, use AddAnnotation instead.");
+    this.argument = argument;
+    this.collapseArguments = collapseArguments;
+  }
+
+  @Override
+  public void visit(NodeWithAnnotations<?> node) {
+
+    StringLiteralExpr argumentExp = new StringLiteralExpr(argument);
+    // Check all existing annotations with arguments.
+    boolean argumentExists =
+        node.getAnnotations().stream()
+            .anyMatch(
+                annot -> {
+                  if (!(annot.getNameAsString().equals(annotation)
+                      || annot.getNameAsString().equals(annotationSimpleName))) {
```

Comment:
Let's not change this in this PR, but it's a bit weird that `annotationSimpleName` (and `annotation`) are fields of `Change`, rather than of `AddAnnotation`.

Reply:
Sure, will do that.

Reply:
#70